core/corehttp/gateway_handler.go

```diff
@@ -334,45 +341,31 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	// set these headers _after_ the error, for we may just not have it
-	// and don't want the client to cache a 500 response...
-	// and only if it's /ipfs!
-	// TODO: break this out when we split /ipfs /ipns routes.
-	modtime := time.Now()
+	// Handling Unixfs
+	dr, err := i.api.Unixfs().Get(r.Context(), resolvedPath)
+	if err != nil {
+		webError(w, "ipfs cat "+escapedURLPath, err, http.StatusNotFound)
+		return
+	}
+	// TODO:  do we want to reuse unixfsGetMetric for block/car, or should we have separate ones?
+	i.unixfsGetMetric.WithLabelValues(parsedPath.Namespace()).Observe(time.Since(begin).Seconds())
```

Comment:
:question: Right now we only calculate time to first block for Unixfs.
Do we need separate metrics for `block` and `car`, or should I  add global `timeToFirstBlockGetMetric`?
@guseggert do you know how `unixfsGetMetric` is used in our infra?